### PR TITLE
asserts: use Fetcher in AddSequenceToUpdate

### PR DESCRIPTION
**Pull request by stolowski**
Use Fetcher in AddSequenceToUpdate. Fix addUnresolvedSeq not returning error.

### CODE REVIEW

File: asserts/pool.go [hunk 1]
```diff
@@ -930,8 +929,24 @@ func (p *Pool) AddSequenceToUpdate(toUpdate *AtSequence, group string) error {
...
-	// sequence forming assertions are never predefined, so no check for it.
-	p.addUnresolvedSeq(&u, gnum)
+	retrieve := func(ref *Ref) (Assertion, error) {
+		return ref.Resolve(p.groundDB.Find)
+	}
+	add := func(a Assertion) error {
+		if !a.Type().SequenceForming() {
+			return p.addUnresolved(a.At(), gnum)
+		}
+		// sequence forming assertions are never predefined, so no check for it.
```
Comment (pedronis) [inline asserts/pool.go:939]: we should also add a comment like

// final add corresponding to toUpdate itself

File: asserts/pool.go [hunk 2]
```diff
@@ -930,8 +929,24 @@ func (p *Pool) AddSequenceToUpdate(toUpdate *AtSequence, group string) error {
...
+		return p.addUnresolvedSeq(&u, gnum)
```
Comment (pedronis) [inline asserts/pool.go:940]: I think for completeness we should do something like:

u.Revision = a.Revision()

and maybe add a test about that

**Review by pedronis — COMMENTED** [review]
looks good, couple small comments

File: asserts/pool.go [hunk 3]
```diff
@@ -936,7 +936,9 @@ func (p *Pool) AddSequenceToUpdate(toUpdate *AtSequence, group string) error {
 		if !a.Type().SequenceForming() {
 			return p.addUnresolved(a.At(), gnum)
 		}
+		u.Revision = a.Revision()
```
Comment (pedronis) [inline asserts/pool.go:939]: nitpick: I would put this after the comments directly before addUnresolvedSeq

**Review by pedronis — APPROVED** [review]
thank you, small nitpick

**Comment by stolowski** [comment]
I accidentally addressed @anonymouse64 's remarks for #9930 here (small test consistency cleanups, small cleanup in addToGrouping), sorry for the mess.

**Comment by pedronis** [comment]
> I accidentally addressed @anonymouse64 's remarks for #9930 here (small test consistency cleanups, small cleanup in addToGrouping), sorry for the mess.

oopsy, @stolowski  it doesn't seem too bad,  could you merge master here now?

**Comment by stolowski** [comment]
> > I accidentally addressed @anonymouse64 's remarks for #9930 here (small test consistency cleanups, small cleanup in addToGrouping), sorry for the mess.
> 
> oopsy, @stolowski it doesn't seem too bad, could you merge master here now?

I already did (not sure why commit history look that way though).

**Comment by stolowski** [comment]
Replaced by #9978

**Pull request closed without merging**